javascript/react/redux-tips/avoid-object-mutations.md

```diff
@@ -30,15 +30,35 @@ aspects:
 
 A **pure** function that returns a new object with the needed modifications can easily be implemented in **Redux**. However, if other properties are added later on to the object, you might need to update several pieces of your code.
 
-To avoid this, the `Object.assign()` method introduced in **ES6** can be used because it returns a new object with all enumerable properties of the *source* object:
+To avoid this, the `Object.assign()` method can be used because it returns a new object with all enumerable properties of the *source* object:
+
+```jsx
+const target = { a: 1, b: 2 };
+const source = { b: 3, c: 4 };
+
+const returnedTarget = Object.assign(
+  target,
+  source
+);
+
+console.log(target);
+// { a: 1, b: 3, c: 4 }
+
+console.log(returnedTarget);
+// { a: 1, b: 3, c: 4 }
+```
+
+This method is used to create a new object by copying the properties from the source object to the target object. This is useful for Redux as the method returns a new object, which keeps in line with creating **pure** functions. When used inside `reducers`, the target object is the one whose properties are going to be assigned. Because we want to keep our function **pure** we will pass an empty object `{ }`. The source object contains the properties that are going to be copied, and we can even add new properties:
+
 ```javascript
 return Object.assign({}, myObj, {
   myProperty: newValue
 });
 ```
-The first argument of this function, `{ }`, is the one whose properties are going to be assigned. As we want to keep our function **pure** we pass an empty `Object`. The others represent objects whose properties are going to be copied. If there are two objects that have different values for the same property, the last one wins.
+Note that if there are two objects that have different values for the same property, the last one wins.
```

Comment:
Maybe elaborate here that the "rightmost" object wins, since "last" isn't as clear

Reply:
New phrase:

Note that if there are two objects that have different values for the same property, the rightmost object wins. Looking at our example, this means that if `myObj` has the `myProperty: oldValue` property it will be replaced by `myProperty: newValue`.